### ISSUE TITLE
Notes won’t open in search results 

### ISSUE THREAD

**Issue by jsd137** [standardnotes/forum#2277]
**Describe the bug**
After searching, I tap a note from the search results list to open it. It won’t open. I long tap it to see details. Close details. Then it will open.  Subsequent searches seem to work ok. 

**Smartphone (please complete the following information):**
 - Device: iphone 13 pro
 - OS:  iOS 15.6.1
 - mobile app new interface v3.45.4 (288)

**Comment by effieeee** [comment]
Hi @jsd137, I'm afraid I couldn't reproduce this on iOS so far, but I wanted to ask how long it takes before the issue recurs on the app? Does it happen after each relaunch of the app?

**Comment by jsd137** [comment]
Hello. It occurs when I first open the app. Do a search. Scroll the search results. And then try to open a note. It seems to be intermittent. I’ll try to figure out the exact situation that causes it.

**Comment by meewgumi** [comment]
This happens with me every time I do a search. I wonder if it's because of the plugins I have installed? It's only happened since the new app version came out, I had no issues on the legacy version and I miss being able to turn off the beta so I could use the app

**Comment by jsd137** [comment]
It’s still happening to me.  I don’t have any plugins installed.  On Nov 21, 2022, at 10:16 PM, Megumi ***@***.***> wrote:﻿
This happens with me every time I do a search. I wonder if it's because of the plugins I have installed? It's only happened since the new app version came out, I had no issues on the legacy version and I miss being able to turn off the beta so I could use the app

**Comment by homberger** [comment]
I am affected by this too. Makes this app useless.

iOS App Version: v3.46.28 (355)

**Comment by homberger** [comment]
As soon as I clicked in the search input field it is not longer possible to view any notes.

**Comment by effieeee** [comment]
A report over email also mentions that this persists even on the v3.114, will have to investigate further

**Comment by homberger** [comment]
Problem persists with new iOS release (2022-12-03).
There is a version mismatch between App Store and in the app itself:
App store version identifier: 3.117.3
Version identifier in the app: v3.116.0 (381)

**Comment by homberger** [comment]
@effieeee
I analysed the traffic of Standard Notes on my iPhone with mitmproxy and maybe I found something interesting.

**The iOS app is setting this request header for at least one type of requests: `origin: file://`.** This header should not be set to `file://`, afaik, and the desktop app does not set this header at all.

This is such an http request of the ios app with request header `Origin: file://`:
```
POST https://notes.example.com/v1/items HTTP/2.0
content-type: application/json
accept: */*
authorization: Bearer 1:XXXXXXXXXXXXXXXXXXXXXXXXXXX
x-application-version: Mobile-3.116.0 (381)
accept-encoding: gzip, deflate, br
accept-language: de-DE,de;q=0.9
origin: file://
user-agent: Mozilla/5.0 (iPhone; CPU iPhone OS 16_1_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148
x-snjs-version: 2.155.0
content-length: 85
```

The response headers `Content-Security-Policy` and `Access-Control-Allow-Origin` of the api backend might block these requests.

The response header `Content-Security-Policy` does not contain the scheme `filesystem:` for any types (eg: default-src, source-src or object-src). 
The response header `Access-Control-Allow-Origin` is set to `*` and does not vary by origin. The asterisk does not include the filesystem origin.

This is the actual response header **Content-Security-Policy**:
`Content-Security-Policy: default-src https: 'self';base-uri 'self';child-src * blob:;connect-src *;font-src * 'self';form-action 'self';frame-ancestors * *.standardnotes.org *.standardnotes.com;frame-src * blob:;img-src 'self' * data:;manifest-src 'self';media-src 'self';object-src 'self';script-src 'self';style-src 'self';script-src-attr 'none';upgrade-insecure-requests`

This might be a solution: Prevent the request header `Origin: file://`.
